### PR TITLE
Upgrade kafka 2.6.0

### PR DESCRIPTION
**Pull request by sunshine69**

### CODE REVIEW

**Comment by sunshine69** [comment]
This has been a request https://github.com/epiphany-platform/epiphany/issues/1137 and we are currently suffering from this kafka for too long.

**Comment by sunshine69** [comment]
Please note that I have tested this PR and successfully build a epiph cluster with kafka version 2.6.0. 

From the original request (not mine) I saw that we are not planned for 0.8.0 but it is not clear to me the reason. Can someone please share a bit more about it? I would like to know what is the technical issues, or procedure issues that prevent us to move forward?

Thanks

**Comment by seriva** [comment]
Hey 

Few minor comments:
- Missing changelog entry
- Update version in component list (https://github.com/epiphany-platform/epiphany/blob/develop/docs/home/COMPONENTS.md#epiphany-cluster-components)

Few bigger things:
- Missing upgrade for existing clusters, this will break existing installations.
- We need to check Zookeeper as well.  If I remember there are some issues regarding compatibility or do we even still need it? (https://github.com/epiphany-platform/epiphany/issues/1136)

Cheers
Luuk

**Comment by sunshine69** [comment]
Thanks Luuk, I will look into it as soon as I can

**Comment by sunshine69** [comment]
@seriva Are we meant to support upgrade for kafka? To my best of knowledge we don't support it even for current - it is complex process.

Re-run epiphany and update component is still a test/try progress atm and depending on many components, some might work, some might not. I saw a page somewhere listing state of it and I remember we mostly don't support it globally. Am I correct?

I will try to look into upgrade as well by the way admiting it would be very complex situation .

**Comment by seriva** [comment]
@sunshine69 Sorry for the late reply but yes this upgrade should be taken into consideration looking at the issue Tomasz made for that #1137 

image

Otherwise existing cluster might have issues.

**Comment by plirglo** [comment]
Hi @sunshine69 

As @seriva mentioned while changing version of component we should also create automation to upgrade this component (by running epicli upgrade).

To see how it was done for upgrade between Kafka 2.0.0 to recent 2.3.1 please look at this part of upgrade code [here](https://github.com/epiphany-platform/epiphany/blob/310b086e62eecd8a8613b723e2bd69d41bf174ea/core/src/epicli/data/common/ansible/playbooks/upgrade.yml#L143) and ansible tasks located [here](https://github.com/epiphany-platform/epiphany/blob/v0.8/core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/kafka.yml).

Also with Kafka 2.6 Apache released new version of Zookeeper [here](https://kafka.apache.org/downloads), so in this case we should also upgrade Zookeeper to keep configuration consistent.

Regards,
Irek

File: core/src/epicli/data/common/ansible/playbooks/roles/repository/files/download-requirements/redhat-7/requirements.txt [hunk 1]
```diff
@@ -145,8 +145,8 @@ kubernetes-cni-0.8.6-0
 
 [files]
 https://github.com/prometheus/haproxy_exporter/releases/download/v0.10.0/haproxy_exporter-0.10.0.linux-amd64.tar.gz
-https://repo1.maven.org/maven2/io/prometheus/jmx/jmx_prometheus_javaagent/0.12.0/jmx_prometheus_javaagent-0.12.0.jar
-https://archive.apache.org/dist/kafka/2.3.1/kafka_2.12-2.3.1.tgz
+https://repo1.maven.org/maven2/io/prometheus/jmx/jmx_prometheus_javaagent/0.14.0/jmx_prometheus_javaagent-0.14.0.jar
+http://apache.mirror.amaze.com.au/kafka/2.6.0/kafka_2.12-2.6.0.tgz
```
Comment (erzetpe) [inline core/src/epicli/data/common/ansible/playbooks/roles/repository/files/download-requirements/redhat-7/requirements.txt:149]: Why you have chosen this mirror?

File: core/src/epicli/data/common/ansible/playbooks/roles/repository/files/download-requirements/ubuntu-18.04/requirements.txt [hunk 2]
```diff
@@ -172,8 +172,8 @@ kubernetes-cni 0.7.5-00
 kubernetes-cni 0.8.6-00
 
 [files]
-https://repo1.maven.org/maven2/io/prometheus/jmx/jmx_prometheus_javaagent/0.12.0/jmx_prometheus_javaagent-0.12.0.jar
-https://archive.apache.org/dist/kafka/2.3.1/kafka_2.12-2.3.1.tgz
+https://repo1.maven.org/maven2/io/prometheus/jmx/jmx_prometheus_javaagent/0.14.0/jmx_prometheus_javaagent-0.14.0.jar
+http://apache.mirror.amaze.com.au/kafka/2.6.0/kafka_2.12-2.6.0.tgz
```
Comment (erzetpe) [inline core/src/epicli/data/common/ansible/playbooks/roles/repository/files/download-requirements/ubuntu-18.04/requirements.txt:176]: Why you have chosen this mirror?

**Comment by atsikham** [comment]
This PR seems to be ok as we already have Kafka part in upgrade role, but Pawel's questions still have no answers. We support only `offline` upgrade and I see no changes to make there.

**Comment by atsikham** [comment]
@toszo so this PR can be closed as implemented in #1890?

**Pull request closed without merging**